pkg/controller/clusterdeployment/clusterdeployment_controller_test.go

```diff
@@ -94,6 +94,8 @@ current-context: admin
 
 func init() {
 	log.SetLevel(log.DebugLevel)
+	// Register the metrics to avoid panics during testing
+	registerMetrics(mapClusterTypeLabelToValue)
```

Comment:
The metrics do not need to be registered, but they have to be defined at least - because code never checks whether a metric is registered, it always attempts to observe it, and prometheus libraries handle the rest.
I've changed the wording a bit, and I do [intend to implement the unit testing in a follow-up PR](https://issues.redhat.com/browse/HIVE-2092)